Voinkova

jacknondaniels opened the pull request:
Type of stored items can be changed in main.

jacknondaniels committed:
queue with certain type of data

jacknondaniels committed:
template queue

jacknondaniels committed:
template queue

Ainoss commented on task02/main.cpp:
```diff
@@ -1,27 +1,63 @@
 
 #include <iostream>
+#include "queue.cpp"
```
It is the wrong pattern, never use it! 
We include header files and compile code (.cpp) files. So put your templates implementation into the header (you could read about template instantiation and code generation to understand details).

Ainoss commented on task02/queue.cpp:
```diff
@@ -0,0 +1,120 @@
+        current_number(0),
+        size (0)
+    {
+        if (!(array = new T [m_size]))
```
It is so sad, but `new` never return NULL! It throws exception instead, so you don't need to throw it by yourself.

Ainoss commented on task02/main.cpp:
```diff
@@ -1,27 +1,63 @@
+                queue->enqueue(d);
+            }
+            else if (c == 't')
+                cout<<queue->dequeue()<<endl;
```
Code style:  we always put spaces around operators! (but not for brackets)

Ainoss commented on task02/queue.cpp:
```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include "queue.h"
+
+template <typename T>  class Note
```
Better place it into ListQueue class declaration.

Ainoss commented on task02/queue.cpp:
```diff
@@ -0,0 +1,120 @@
+        if (head)
+        {
+            Note<T> *current = head;
+            while (current->getnext())
```
This is OK for little homework, but for real Queue, it is so slow. It is O(n) complexity.
Better to have a pointer to another end of the list. It will have O(1).

Ainoss commented on task02/queue.cpp:
```diff
@@ -0,0 +1,120 @@
+    T *array;
+    unsigned m_size;
+    unsigned current_number;
+    unsigned size;
```
Confusing naming, we have m_size and size... But both are members... One is const by nature, but other is not. Too complex!

Ainoss reviewed (commented):
Good start! 
But this solution needs some rework. Follow comments:

jacknondaniels committed:
template implementation made in a header. Class Note definition made inside of class ListQueue. Pointer to the last member of a ListQueue added.

jacknondaniels commented:
Some commits were made! I tried to consider all things mentioned. Sorry for being late.

Ainoss committed:
Add STL list implementation (9 lesson)

Just a little study example

Ainoss committed:
Add task 3: FileIndexer implementation

This task covers some STL containers and basic smart pointers usage.

Ainoss committed:
Add task 4: Storage class specialization

In this task we need class template specialization for char*

Ainoss committed:
Fix headers in task 4

Add platform-specific header ifdef to support compilation on Windows.

jacknondaniels committed:
Merge branch 'master' of https://github.com/Ainoss/dgap-cpp-course into voinkova

jacknondaniels committed:
Task 4: specialization for a template

jacknondaniels committed:
Task 4: specialization for a template

jacknondaniels committed:
First working version of FileIndexer;

jacknondaniels committed:
First working version of FileIndexer;

jacknondaniels committed:
The most Duck version;

jacknondaniels committed:
The most Duck version;

jacknondaniels committed:
Program doesn't quack any more.
Definitions of used constants were added.
Output is table-view.
Mechanism for sorting is 'from least to biggest' for size, 'from newest to oldest' for time.
Printing function is made inline in order to avoid either code duplicating or running-time wasting.

jacknondaniels committed:
Merge remote-tracking branch 'origin/voinkova' into voinkova

# Conflicts:
#	task03/index.cpp

Ainoss commented:
Задание 2 принято, отличная работа.

Ainoss commented:
Закрываю, т.к. вижу остальные задания в отдельном PR
